File: backend/app/schemas/fiscal_reference.py
from __future__ import annotations

from decimal import Decimal

from pydantic import BaseModel


class FiscalReferenceEntryResponse(BaseModel):
    source_kind: str
    source_sheet: str
    source_row_number: int
    cnae: str | None = None
    cnae_formatted: str | None = None
    cnae_description: str | None = None
    lc116_item: str | None = None
    lc116_item_formatted: str | None = None
    lc116_description: str | None = None
    iss_rate: Decimal | None = None
    allows_tax_outside: bool | None = None
    lc116_inciso: str | None = None
    trib_nac_code: str | None = None
    trib_nac_description: str | None = None
    nbs: str | None = None
    nbs_formatted: str | None = None
    nbs_description: str | None = None
    class_trib_code: str | None = None
    class_trib_description: str | None = None
    cst_code: str | None = None
    cst_description: str | None = None
    indop_code: str | None = None
    indop_description: str | None = None
    onerous: bool | None = None
    foreign_acquisition: bool | None = None
    ibs_incidence_location: str | None = None


class FiscalReferenceSourceResponse(BaseModel):
    source_kind: str
    source_title: str
    source_version: str | None = None
    source_file_name: str
    file_sha256: str
    imported_at: str


class FiscalReferenceSummaryResponse(BaseModel):
    cnaes: int
    lc116_items: int
    nbs_codes: int
    reference_rows: int


class FiscalReferenceSearchResponse(BaseModel):
    search_type: str
    query: str
    normalized_query: str
    summary: FiscalReferenceSummaryResponse
    cnae_lc116: list[FiscalReferenceEntryResponse]
    lc116_nbs: list[FiscalReferenceEntryResponse]
    tribnac_nbs: list[FiscalReferenceEntryResponse]
    sources: list[FiscalReferenceSourceResponse]

File: frontend/src/types/fiscalReference.ts
export type FiscalReferenceEntry = {
  source_kind: string;
  source_sheet: string;
  source_row_number: number;
  cnae: string | null;
  cnae_formatted: string | null;
  cnae_description: string | null;
  lc116_item: string | null;
  lc116_item_formatted: string | null;
  lc116_description: string | null;
  iss_rate: string | null;
  allows_tax_outside: boolean | null;
  lc116_inciso: string | null;
  trib_nac_code: string | null;
  trib_nac_description: string | null;
  nbs: string | null;
  nbs_formatted: string | null;
  nbs_description: string | null;
  class_trib_code: string | null;
  class_trib_description: string | null;
  cst_code: string | null;
  cst_description: string | null;
  indop_code: string | null;
  indop_description: string | null;
  onerous: boolean | null;
  foreign_acquisition: boolean | null;
  ibs_incidence_location: string | null;
};

export type FiscalReferenceSearchResponse = {
  search_type: "cnae" | "nbs";
  query: string;
  normalized_query: string;
  summary: { cnaes: number; lc116_items: number; nbs_codes: number; reference_rows: number };
  cnae_lc116: FiscalReferenceEntry[];
  lc116_nbs: FiscalReferenceEntry[];
  tribnac_nbs: FiscalReferenceEntry[];
  sources: Array<{ source_kind: string; source_title: string; source_version: string | null; source_file_name: string; file_sha256: string; imported_at: string }>;
};
